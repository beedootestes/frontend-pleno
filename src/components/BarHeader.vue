<template>
  <div class="bar-header">
    <img class="beedoo-logo" src="../assets/beedoo-logo.png" alt="Beedoo logo" />

    <div class="cart-profile">
      <div class="cart-container">
        <CartIcon :size="35" />
      </div>
      <figure class="profile-image">
        <img src="/profile.jpg" alt="Imagem perfil" />
      </figure>
    </div>
  </div>
</template>

<script setup lang="ts">
import CartIcon from 'vue-material-design-icons/Cart.vue';
</script>

<style scoped lang="scss">
  .bar-header {
    display: flex;
    width: 100%;
    align-items: center;
    justify-content: space-between;

    .beedoo-logo {
      display: block;
      height: 54px;
    }

    .cart-profile {
      display: flex;

      .cart-container {
        border-radius: 100%;
        box-sizing: border-box;
        padding: 13px;
        display: flex;
        align-items: center;
        background-color: $color-inactive;
        margin-right: 15px;
      }

      .profile-image {
        height: 65px;
        width: 65px;
        box-sizing: border-box;
        border-radius: 100%;
        margin: unset;

        > img {
          border-radius: 100%;
          height: 65px;
          width: 65px;
          object-fit: cover;
          display: block;
          border: 2px solid $color-inactive;
        }
      }
    }
  }
</style>