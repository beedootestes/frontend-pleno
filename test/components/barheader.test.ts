import { describe, expect, test } from "vitest";
import BarHeader from "../../src/components/BarHeader.vue";
import { mount, VueWrapper } from "@vue/test-utils";


describe('Test BarHeader.vue', () => {
  test('mount component', async () => {
    expect(BarHeader).toBeTruthy()

    const barheader: VueWrapper<BarHeader> = mount(BarHeader)

    expect(barheader.html()).toMatchSnapshot()
  })
})
