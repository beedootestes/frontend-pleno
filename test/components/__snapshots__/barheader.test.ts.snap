// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test BarHeader.vue > mount component 1`] = `
"<div data-v-3471366b=\\"\\" class=\\"bar-header\\"><img data-v-3471366b=\\"\\" class=\\"beedoo-logo\\" src=\\"/src/assets/beedoo-logo.png\\" alt=\\"Beedoo logo\\">
  <div data-v-3471366b=\\"\\" class=\\"cart-profile\\">
    <div data-v-3471366b=\\"\\" class=\\"cart-container\\"><span data-v-3471366b=\\"\\" aria-hidden=\\"true\\" class=\\"material-design-icon cart-icon\\" role=\\"img\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"35\\" height=\\"35\\" viewBox=\\"0 0 24 24\\"><path d=\\"M17,18C15.89,18 15,18.89 15,20A2,2 0 0,0 17,22A2,2 0 0,0 19,20C19,18.89 18.1,18 17,18M1,2V4H3L6.6,11.59L5.24,14.04C5.09,14.32 5,14.65 5,15A2,2 0 0,0 7,17H19V15H7.42A0.25,0.25 0 0,1 7.17,14.75C7.17,14.7 7.18,14.66 7.2,14.63L8.1,13H15.55C16.3,13 16.96,12.58 17.3,11.97L20.88,5.5C20.95,5.34 21,5.17 21,5A1,1 0 0,0 20,4H5.21L4.27,2M7,18C5.89,18 5,18.89 5,20A2,2 0 0,0 7,22A2,2 0 0,0 9,20C9,18.89 8.1,18 7,18Z\\"><!--v-if--></path></svg></span></div>
    <figure data-v-3471366b=\\"\\" class=\\"profile-image\\"><img data-v-3471366b=\\"\\" src=\\"/profile.jpg\\" alt=\\"Imagem perfil\\"></figure>
  </div>
</div>"
`;
